core/whistle_transactions-1.0.0/src/wh_transaction.erl

```diff
@@ -75,159 +71,98 @@
+-spec sub_account_id(transaction()) -> ne_binary().
 sub_account_id(#wh_transaction{sub_account_id=SubAccountId}) ->
     SubAccountId.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
+-spec event(transaction()) -> ne_binary().
 event(#wh_transaction{event=Event}) ->
     Event.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
+-spec number(transaction()) -> ne_binary().
 number(#wh_transaction{number=Number}) ->
     Number.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
+-spec feature(transaction()) -> ne_binary().
 feature(#wh_transaction{feature=Feature}) ->
     Feature.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
+-spec bookkeeper_info(transaction()) -> wh_json:object().
 bookkeeper_info(#wh_transaction{bookkeeper_info=BookkeeperInfo}) ->
     BookkeeperInfo.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
+-spec metadata(transaction()) -> wh_json:object().
 metadata(#wh_transaction{metadata=MetaData}) ->
     MetaData.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
+-spec reason(transaction()) -> ne_binary().
 reason(#wh_transaction{pvt_reason=Reason}) ->
     Reason.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
+-spec code(transaction()) -> non_neg_integer().
 code(#wh_transaction{pvt_code=Code}) ->
     Code.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
+-spec amount(transaction()) -> non_neg_integer().
 amount(#wh_transaction{pvt_amount=Amount}) ->
     Amount.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
+-spec type(transaction()) -> ne_binary().
 type(#wh_transaction{pvt_type=Type}) ->
     Type.
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
+-spec created(transaction()) -> wh_now().
```

Comment:
`wh_now()` is the 3-tuple from `erlang:now()`. If you look at `wh_util:current_tstamp()`, the return is `gregorian_seconds()`; use that instead.